src/Diagnostics.RuntimeHost/Utilities/RewriteDiagnosticResource.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using Microsoft.AspNetCore.Rewrite;
+using Diagnostics.Logger;
+using System.Linq;
+using System.Net;
+using Microsoft.AspNetCore.Http;
+using Diagnostics.RuntimeHost.Utilities;
+
+namespace Diagnostics.RuntimeHost.Utilities
+{
+    public class RewriteDiagnosticResource : IRule
+    {
+        public int StatusCode { get; } = (int)HttpStatusCode.BadRequest;
+
+        public void ApplyRule(RewriteContext context)
+        {
+            var request = context.HttpContext.Request;
+            // Partner teams using Diagnose and Solve, make the request to api/invoke from their RP.
+            if (!request.Path.Value.Equals(UriElements.PassThroughAPIRoute, StringComparison.OrdinalIgnoreCase))
+            {
+                context.Result = RuleResult.ContinueRules;
+                return;
+            } else
+            {
+                // Check for required headers and send 400 if not present.
+                if (!request.Headers.TryGetValue(HeaderConstants.ApiPathHeader, out var apiPaths) || !apiPaths.Any() || string.IsNullOrWhiteSpace(apiPaths.FirstOrDefault()))
+                {
+                    var response = context.HttpContext.Response;
+                    response.StatusCode = this.StatusCode;
+                    response.ContentType = "text/plain";
+                    response.WriteAsync($"Missing {HeaderConstants.ApiPathHeader} header").Wait();
```

Comment:
`Wait` would be blocking the request thread. Maybe instead you can write text to memory stream and assign that to `response.Body`